### PR TITLE
feat(instance): seperate root volume and additional volumes

### PR DESCRIPTION
**Pull request by kindermoumoute**

### CODE REVIEW

File: internal/namespaces/instance/v1/command_server_create.go [hunk 1]
```diff
@@ -200,21 +205,21 @@ func instanceServerCreateRun(ctx context.Context, argsI interface{}) (i interfac
...
 		// Create initial volume template map.
-		volumes, err := createInitialVolumeMap(apiInstance, args.Zone, organizationID, args.Volumes)
+		volumes, err := buildVolumes(apiInstance, args.Zone, organizationID, serverReq.Name, args.RootVolume, args.AdditionalVolumes)
 		if err != nil {
 			return nil, err
 		}
 
 		// Validate root volume type and size.
-		if err := validateRootVolume(apiInstance, args.Zone, serverReq.Image, volumes[0]); err != nil {
+		if err := validateRootVolume(apiInstance, args.Zone, serverReq.Image, volumes); err != nil {
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/command_server_create.go]: why do we need all `volumes` to only validate the root volume ?

File: internal/namespaces/instance/v1/command_server_create.go [hunk 2]
```diff
@@ -381,7 +400,11 @@ func buildVolumeTemplate(api *instance.API, zone scw.Zone, orgID, flagV string)
 		return buildVolumeTemplateFromUUID(api, zone, parts[0])
 	}
 
-	return nil, fmt.Errorf("Invalid volume format '%s'.", flagV) // TODO: improve msg [APIGW-1371]
+	return nil, &core.CliError{
+		Err:     fmt.Errorf("Invalid volume format '%s'.", flagV),
+		Details: "",
+		Hint:    `You must provide either a UUID ("7a892c1a-bbdc-491f-9974-4008e3708664"), a local volume size ("local:100G" or "l:100G") or a block volume size ("block:100G" or "b:100G").`,
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/command_server_create.go]: we should make it explicit this is an example UUID

use `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`

File: internal/namespaces/instance/v1/command_server_create.go [hunk 3]
```diff
@@ -431,17 +454,20 @@ func validateLocalVolumeSizes(api *instance.API, volumes []*instance.VolumeTempl
 	}
 
 	// Validate total size.
-	var vConstr *instance.ServerTypeVolumeConstraintSizes
-	if st, exists := serverTypesRes.Servers[commercialType]; exists {
-		vConstr = st.VolumesConstraint
+	if serverType, exists := serverTypesRes.Servers[commercialType]; exists {
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/command_server_create.go]: `exists` -> `serverExists` ?

File: internal/namespaces/instance/v1/command_server_create.go [hunk 4]
```diff
@@ -431,17 +454,20 @@ func validateLocalVolumeSizes(api *instance.API, volumes []*instance.VolumeTempl
...
+		volumeConstraint := serverType.VolumesConstraint
 
-		if localVolumeTotalSize < vConstr.MinSize || localVolumeTotalSize > vConstr.MaxSize {
-			min := humanize.Bytes(uint64(vConstr.MinSize))
-			if vConstr.MinSize == vConstr.MaxSize {
+		if rootVolume, exist := volumes["0"]; !exist || rootVolume == nil {
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/command_server_create.go]: `exists` -> `rootVolumeExists` ?

File: internal/namespaces/instance/v1/command_server_create_test.go [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+import (
+	"testing"
+
+	"github.com/scaleway/scaleway-cli/internal/core"
+)
+
+func Test_CreateServer(t *testing.T) {
+
+	//t.Run("Simple", core.Test(&core.TestConfig{
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/command_server_create_test.go]: remove, uncomment, or add TODO

File: internal/namespaces/instance/v1/command_server_create_test.go [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+		Check: core.TestCheckCombine(
+			core.TestCheckGolden(),
+			core.TestCheckExitCode(1),
+		),
+	}))
+
+	t.Run("Error: invalid image label", core.Test(&core.TestConfig{
+		Parallel: true,
+		Commands: GetCommands(),
+		Cmd:      "scw instance server create image=macos",
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/command_server_create_test.go]: maybe use an os name that doesn't exist at all

what if someday we support macos ?

File: internal/namespaces/instance/v1/command_server_create.go [hunk 7]
```diff
@@ -381,7 +400,11 @@ func buildVolumeTemplate(api *instance.API, zone scw.Zone, orgID, flagV string)
 		return buildVolumeTemplateFromUUID(api, zone, parts[0])
 	}
 
-	return nil, fmt.Errorf("Invalid volume format '%s'.", flagV) // TODO: improve msg [APIGW-1371]
+	return nil, &core.CliError{
+		Err:     fmt.Errorf("Invalid volume format '%s'.", flagV),
```
Comment (QuentinBrosse) [inline internal/namespaces/instance/v1/command_server_create.go]: Remove uppercase first letter and dot.

File: internal/namespaces/instance/v1/command_server_create.go (hunk 4 above)
Comment (QuentinBrosse) [inline internal/namespaces/instance/v1/command_server_create.go]: Maybe add a comment like:
`// If no root volume provided, count the default root volume size added by the API.`

File: internal/namespaces/instance/v1/command_server_create_test.go [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+	//		ctx.ExecuteCmd("scw instance server delete server-id={{ .Server.id }}")
+	//		return nil
+	//	},
+	//	Check: core.TestCheckCombine(
+	//		core.TestCheckGolden(),
+	//	),
+	//}))
+}
+
+func Test_CreateServerErrors(t *testing.T) {
```
Comment (QuentinBrosse) [inline internal/namespaces/instance/v1/command_server_create_test.go]: If I understand well, you never remove created instances?

Comment (kindermoumoute) [reply internal/namespaces/instance/v1/command_server_create_test.go]: None of this test succeed to create an instance. (the test fail if it does)

**Pull request merged**